### DISCUSSION TITLE
5.18 up for testing 🧪

### DISCUSSION THREAD

**Discussion by anthraxx** [anthraxx/linux-hardened#70] [General]
Hey folks,

I've just pushed the `5.18` testing branch. Please feel free to give it a try and come back with some feedback.

**Comment by hardfalcon** [comment]
5.18.x crashes on boot on two of my machines if I don't add either `snd_hda_intel.snoop=1` or `modprobe.blacklist=snd_hda_intel` in the bootloader. The crash occurs also with a vanilla kernel 5.18 if I build the kernel using a hardened config. I'm currently bisecting this, and so far, I've been able to narrow it down to between commit 169e77764adc041b1dacba84ea90516a895d43b2 ("good", does not crash) and 3123109284176b1532874591f7c81f3837bbdc17 ("bad", does crash). So while the bug is not present in linux 5.17, it was introduced somewhere before 5.18-rc1.

Interestingly, both affected machines use dedicated AMD GPUs, and for one of those machines, I also have an otherwise identical machine with a dedicated NVidia GPU that is *not* affected by the crashes. Still, booting with `modprobe.backlist=amdgpu` does not prevent the crashes. Also, one of the affected machines has an Intel Haswell-generation CPU with an integrated GPU (though it's not used and i915 does not get loaded AFAICT), whilst the other uses a Broadwell-generation Xeon without integrated GPU.

I'll keep you posted when I've found out more.

**Comment by hardfalcon** [comment]
Commit 9bf3fc5007856d5ca76d6e7a4652e3b67b683241 is good, too.

**Comment by hardfalcon** [comment]
Narrowed it down to between dfdc1de64248b5e1024d8188aeaf0e59ec6cecd5 (good/no crashes) and 74164d284b2909de0ba13518cc063e9ea9334749 (bad/crashes).

**Comment by hardfalcon** [comment]
Narrowed it down to between 0a815d0135f1be20c1a04d07f7573a26272ef846 (good/no crashes) and 9ae2a143081fa8fba5042431007b33d9a855b7a2 (bad/crashes).

**Reply by hardfalcon** [reply]
Narrowed it down to 1930a6e739c4b4a654a69164dbe39e554d228915 (good/no crashes) and 9ae2a143081fa8fba5042431007b33d9a855b7a2 (bad(crashes).

However, I'm afraid I can't bisect further since all commits within the remaining range that I try seem to throw compiler errors… :/

**Comment by hardfalcon** [comment]
Narrowed it down to 37fcacb50be7071d146144a6c5c5bf0194b9a1cf (good/no crashes) and 9ae2a143081fa8fba5042431007b33d9a855b7a2 (bad/crashes), any further bisecting results in compiler errors.

[This](https://github.com/anthraxx/linux-hardened/files/8850118/config.txt) is the config I've used for compiling the kernel (without the `linux-hardened` patchset applied!).

The relevant difference appear to be the following dmesg excerpts:
`5.17.0-1-hardened-12883-g37fcacb50be7` (booted without `snd_hda_intel.snoop` parameter):
```
snd_hda_intel 0000:00:1b.0: enabling device (0000 -> 0002)
snd_hda_intel 0000:01:00.1: Force to non-snoop mode
```

`5.17.0-1-hardened-12897-g9ae2a143081f` (booted with `snd_hda_intel.snoop=1` to prevent crash):
```
snd_hda_intel 0000:00:1b.0: enabling device (0000 -> 0002)
snd_hda_intel 0000:00:1b.0: Force to snoop mode by module option
snd_hda_intel 0000:01:00.1: Force to snoop mode by module option
```

Relevant lines from `lspci -nn`:
```
00:1b.0 Audio device [0403]: Intel Corporation 8 Series/C220 Series Chipset High Definition Audio Controller [8086:8c20] (rev 04)
01:00.0 VGA compatible controller [0300]: Advanced Micro Devices, Inc. [AMD/ATI] Lexa PRO [Radeon 540/540X/550/550X / RX 540X/550/550X] [1002:699f] (rev c7)
01:00.1 Audio device [0403]: Advanced Micro Devices, Inc. [AMD/ATI] Baffin HDMI/DP Audio [Radeon RX 550 640SP / RX 560/560X] [1002:aae0]
```

**Reply by anthraxx** [reply]
Maybe it would be worth reporting this upstream, so far the scope we have reduced here should be relatively good for the subsystem maintainers to narrow down the actual issue.

As far as I see, this is the last blocker to officially release 5.18, however this isn't really strictly speaking a hardened blocker but a vanilla issue.

**Reply by hardfalcon** [reply]
Reported upstream:
https://bugzilla.kernel.org/show_bug.cgi?id=216112

**Reply by anthraxx** [reply]
Fixed by https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=9882d63bea14c8b3ed2c9360b9ab9f0e2f64ae2b
